src/test/java/net/snowflake/client/jdbc/PreparedStatement1IT.java

```diff
@@ -790,4 +791,46 @@ public void manualTestForPreparedStatementLogging() throws SQLException {
       }
     }
   }
+
+  /**
+   * This test shows how to use batch requests with variant columns. Syntax <code>
+   * insert into ... Select column1, ... from values (?,...)</code> is crucial there.
+   * CLIENT_STAGE_ARRAY_BINDING_THRESHOLD parameter is used on session to set the minimal number of
+   * bind parameters to create stage file with provided parameters.
+   */
+  @Test
+  public void shouldUseVariantBindParametersSNOW1531307() throws SQLException {
```

Comment:
If we want to give a good example of binding variants, it may be good to provide more types than objects, maybe at least a string, a number, an object and and array?

Reply:
As discussed outside of gh - I added case for array, other tests will receive samples probably in the future